### ISSUE TITLE
Why LeViT needs 1000 training epochs?

### ISSUE THREAD

**Issue by lilujunai** [facebookresearch/LeViT#3]
While other VIT models are trained with only 300 epochs, LeViT need 1000 epochs，which bring lots of traing cost. I think is unfair for comparison. What is the accuracy of LeViT at 300 epoch ?

**Comment by btgraham** [comment]
We compare to DeiT models trained for 1000 epochs for fair comparison. We also tried training EfficientNet with distillation, but it did not improve over the baseline accuracy.

The published LeViT models are trained with a large effective batch size (32 nodes*256 images/node). I suspect with a smaller batchsize, similar results can be obtained in 300 epochs, but haven't had the chance to try it.

**Comment by Openning07** [comment]
Thanks for sharing your code!
I've tried to train LeViT-128S with with a smaller batch-size, i.e., 2,048, due to the limited GPU resource, and neglect the distillation head. Then, it comes out with 63% Top-1 acc at Epoch 100 and 66% Top-1 acc at Epoch 240.
So it seems that 2,048 is an ineffective batch-size ?

Would you mind also share your training log to make it easier for us to reproduce the results? Thank~

**Comment by Openning07** [comment]
> While other VIT models are trained with only 300 epochs, LeViT need 1000 epochs，which bring lots of traing cost. I think is unfair for comparison. What is the accuracy of LeViT at 300 epoch ?

Since the author trained LeViT models with a batch-size of 8,192, it means that compared with a batch-size of 2,048 (a common batch-size in ViT models), LeViT conducts less backward propagations, about 4 times less that the original one. If we change the view from *Training Epoch* to *Training Iteration*, then LeViT uses even less iterations to reaches similar classification performance...

**Comment by btgraham** [comment]
Test accuracy (without fusing Conv2D+BatchNorm, etc) every 20 epochs for LeViT-128S during training goes:
* Acc@1 51.749 Acc@5 76.192 loss 2.164
* Acc@1 61.438 Acc@5 83.649 loss 1.685
* Acc@1 63.057 Acc@5 84.542 loss 1.611
* Acc@1 63.944 Acc@5 85.186 loss 1.557
* Acc@1 65.050 Acc@5 86.331 loss 1.501
* Acc@1 65.301 Acc@5 86.250 loss 1.488
* Acc@1 65.337 Acc@5 86.232 loss 1.482
* Acc@1 65.458 Acc@5 86.342 loss 1.488
* Acc@1 66.826 Acc@5 87.079 loss 1.434
* Acc@1 66.601 Acc@5 87.117 loss 1.417
* Acc@1 66.762 Acc@5 87.319 loss 1.418
* Acc@1 67.424 Acc@5 87.599 loss 1.379
* Acc@1 67.779 Acc@5 87.599 loss 1.379
* Acc@1 68.298 Acc@5 88.299 loss 1.347
* Acc@1 67.299 Acc@5 87.388 loss 1.407
* Acc@1 67.537 Acc@5 87.621 loss 1.398
* Acc@1 67.257 Acc@5 87.357 loss 1.400
* Acc@1 68.905 Acc@5 88.472 loss 1.318
* Acc@1 68.701 Acc@5 88.418 loss 1.324
* Acc@1 68.987 Acc@5 88.532 loss 1.318
* Acc@1 68.805 Acc@5 88.665 loss 1.331
* Acc@1 69.622 Acc@5 88.950 loss 1.282
* Acc@1 69.677 Acc@5 89.055 loss 1.276
* Acc@1 69.314 Acc@5 88.785 loss 1.290
* Acc@1 69.616 Acc@5 88.734 loss 1.298
* Acc@1 70.589 Acc@5 89.669 loss 1.228
* Acc@1 70.832 Acc@5 89.518 loss 1.230
* Acc@1 70.742 Acc@5 89.444 loss 1.231
* Acc@1 71.311 Acc@5 89.991 loss 1.200
* Acc@1 71.539 Acc@5 90.202 loss 1.186
* Acc@1 71.580 Acc@5 90.279 loss 1.191
* Acc@1 72.395 Acc@5 90.682 loss 1.145
* Acc@1 72.922 Acc@5 90.905 loss 1.124
* Acc@1 72.844 Acc@5 90.926 loss 1.129
* Acc@1 73.223 Acc@5 91.053 loss 1.111
* Acc@1 73.436 Acc@5 91.256 loss 1.099
* Acc@1 73.683 Acc@5 91.509 loss 1.079
* Acc@1 73.547 Acc@5 91.379 loss 1.091
* Acc@1 74.257 Acc@5 91.656 loss 1.064
* Acc@1 74.572 Acc@5 91.801 loss 1.055
* Acc@1 75.193 Acc@5 92.131 loss 1.017
* Acc@1 75.457 Acc@5 92.248 loss 1.012
* Acc@1 75.611 Acc@5 92.454 loss 0.994
* Acc@1 75.894 Acc@5 92.600 loss 0.985
* Acc@1 76.115 Acc@5 92.603 loss 0.975
* Acc@1 76.226 Acc@5 92.759 loss 0.974
* Acc@1 76.409 Acc@5 92.839 loss 0.968
* Acc@1 76.510 Acc@5 92.883 loss 0.961
* Acc@1 76.445 Acc@5 92.877 loss 0.959
* Acc@1 76.514 Acc@5 92.878 loss 0.958
